servers/src/test/java/tachyon/worker/lineage/LineageDataManagerTest.java

```diff
@@ -0,0 +1,113 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.lineage;
+
+import java.io.OutputStream;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.google.common.collect.Lists;
+
+import tachyon.Constants;
+import tachyon.Sessions;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.FileInfo;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.util.io.BufferUtils;
+import tachyon.util.io.PathUtils;
+import tachyon.worker.block.BlockDataManager;
+import tachyon.worker.block.io.BlockReader;
+
+/**
+ * Tests {@link LineageDataManager}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({BlockDataManager.class, BufferUtils.class})
+public final class LineageDataManagerTest {
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void persistFileTest() throws Exception {
+    long fileId = 1;
+    List<Long> blockIds = Lists.newArrayList(1L, 2L);
+
+    // mock block data manager
+    BlockDataManager blockDataManager = Mockito.mock(BlockDataManager.class);
+    FileInfo fileInfo = new FileInfo();
+    fileInfo.path = "test";
+    Mockito.when(blockDataManager.getFileInfo(fileId)).thenReturn(fileInfo);
+    BlockReader reader = Mockito.mock(BlockReader.class);
+    for (long blockId : blockIds) {
+      Mockito.when(blockDataManager.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId))
+          .thenReturn(blockId);
+      Mockito
+          .when(blockDataManager.readBlockRemote(Sessions.CHECKPOINT_SESSION_ID, blockId, blockId))
+          .thenReturn(reader);
+    }
+
+    LineageDataManager manager = new LineageDataManager(blockDataManager);
+
+    // mock ufs
+    UnderFileSystem ufs = Mockito.mock(UnderFileSystem.class);
+    String ufsRoot = new TachyonConf().get(Constants.UNDERFS_ADDRESS);
+    Mockito.when(ufs.exists(ufsRoot)).thenReturn(true);
+    Whitebox.setInternalState(manager, "mUfs", ufs);
+    OutputStream outputStream = Mockito.mock(OutputStream.class);
+
+    // mock BufferUtils
+    PowerMockito.mockStatic(BufferUtils.class);
+
+    String dstPath = PathUtils.concatPath(ufsRoot, fileInfo.getPath());
+    Mockito.when(ufs.create(dstPath)).thenReturn(outputStream);
+
+    manager.persistFile(fileId, blockIds);
+
+    // verify file persisted
+    List<Long> persistedFiles = (List<Long>) Whitebox.getInternalState(manager, "mPersistedFiles");
+    Assert.assertEquals(Lists.newArrayList(fileId), persistedFiles);
+
+    // verify fastCopy called twice
+    PowerMockito.verifyStatic(Mockito.times(2));
```

Comment:
why verifying `fastCopy` called twice? This is not obviously the expected behavior to unit-test `persistedFile`.